components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -671,7 +671,14 @@ def canChgrp(self):
         group. Web client will only allow this for the data Owner. Admin CAN
         move other user's data, but we don't support this in Web yet.
         """
-        return self.isOwned() or self._conn.isAdmin()   # See #8974
+        return self.getDetails().getPermissions().canChgrp()
```

Comment:
Interesting point. The previous condition is,
```python
self.isOwned() or self._conn.isAdmin()
```
and I've documented `Chgrp2` as, "The user must be either an administrator, or the owner of the objects and a member of the target group."

While `canChgrp` will return `false` for things like `FilterSet` or `DatasetImageLink` (wants the `Dataset` or `Image` selected) it should return exactly as the previous condition for the kinds of things that are selectable by the user in the web. So seeing as CI looks fairly happy I'd guess we're probably safe; I wonder if there's anything else specific we should test.